Tests/iaas/flavor-naming/flavor-add-extra-specs.py

```diff
@@ -298,64 +208,77 @@ def main(argv):
-        if DEBUG:
-            print(f"DEBUG: Inspecting flavor {flavor.name}/{flname} ...")
-        try:
-            flvnm = parser_v2(flname)
-        except ValueError as exc:
+        extra_names_to_check = [
+            (key, value)
+            for key, value in flavor.extra_specs.items()
+            if scs_name_pattern.match(key)
+        ]
+        names_to_check = [('name', flavor.name)] if flavor.name.startswith('SCS-') else []
+        names_to_check.extend(extra_names_to_check)
+
+        # syntax check: compute flavorname instances
+        flavornames = {}
+        for key, name_str in names_to_check:
             try:
-                flvnm = parser_v1(flname)
-                is_v1 = True
-            except ValueError:
-                print(f"ERROR with flavor {flavor.name}: {exc!r}, skipping ...",
-                      file=sys.stderr)
+                flavornames[key] = parser_vN(name_str)
+            except ValueError as exc:
+                logger.error(f"could not parse {key}={name_str}: {exc!r}")
                 errors += 1
-                continue
 
-        # Now do sanity checks (std properties)
-        stderrs = check_std_props(flavor, flvnm, " by name")
-        errors += stderrs
-        if stderrs:
-            continue
-
-        # Parse and Generate name-v1 and name-v2
-        upd = check_name_extra(flavor, "v2", not is_v1, flname)
-        if upd:
-            errors += update_flavor_extra(compute, flavor, "scs:name-v2")
-            chg += 1
-        upd = check_name_extra(flavor, "v1", is_v1, flname)
-        if upd:
-            errors += update_flavor_extra(compute, flavor, "scs:name-v1")
-            chg += 1
-        # Parse and Generate cpu-type and disk0-type
-        upd = check_extra_type(flavor, "cpu-type", flvnm.cpuram.cputype, CPUTYPE_KEY)
-        if upd:
-            errors += update_flavor_extra(compute, flavor, "scs:cpu-type")
-            chg += 1
-
-        # We may not have a disk (or the type is unknown)
-        if flvnm.disk:
-            dtp = flvnm.disk.disktype
-            if not dtp:
-                dtp = disk0_type
+        # select a reference flavorname, check flavornames
+        if flavornames:
+            flavorname_items = iter(flavornames.items())
+            reference_key, reference = next(flavorname_items)
+            reference_core = _extract_core(reference)
+            # sanity check: claims must be true wrt actual flavor
+            errors += check_std_props(flavor, reference, " by name")
+            # sanity check: claims must coincide (check remaining flavornames)
+            for key, flavorname in flavorname_items:
+                errors += check_std_props(flavor, flavorname, " by name")
+                core = _extract_core(flavorname)
+                if core != reference_core:
+                    logger.error(f"Inconsistent {key} vs. {reference_key}: {core} vs. {reference_core}")
```

Comment:
I think we can't output an error here because understatement is allowed. For instance, if we find a name that says the cpu is a dedicated core and one name that says the cpu is a crowded core, then we have to believe the former one and assume the latter one is an understatement. We could still output a warning though.

True inconsistencies would be something such as cpu architecture Intel vs. AMD.

Reply:
We have the same problem when setting the values for `scs:cpu-type` – I think we have to consider all flavornames, not just the "reference", in order to find a "maximum".

Reply:
Or maybe the maximum is not even desired? Maybe the user must disambiguate.

Reply:
I changed it to a warning.